.credo.exs

```diff
@@ -7,6 +7,7 @@
         excluded: []
       },
       checks: [
+        {Credo.Check.Readability.ParenthesesOnZeroArityDefs, [parens: true]},
```

Comment:
Why do we need this? It's a common approach to skip this

Reply:
Credo enforces either we use ParenthesesOnZeroArity or we don't. Since using them required less changes to the codebase, I went and set parens to true.

Reply:
if I comment this out I get
```
┃ [R] ↘ Do not use parentheses when defining a function which has no arguments.
┃       lib/ethereumex/config.ex:5 #(Ethereumex.Config.setup_children)
┃ [R] ↘ Do not use parentheses when defining a function which has no arguments.
┃       lib/ethereumex/counter.ex:8 #(Ethereumex.Counter.setup)
┃ [R] ↘ Do not use parentheses when defining a function which has no arguments.
┃       lib/ethereumex/config.ex:87 #(Ethereumex.Config.ipc_request_timeout)
┃ [R] ↘ Do not use parentheses when defining a function which has no arguments.
┃       lib/ethereumex/config.ex:82 #(Ethereumex.Config.ipc_max_worker_overflow)
┃ [R] ↘ Do not use parentheses when defining a function which has no arguments.
┃       lib/ethereumex/config.ex:77 #(Ethereumex.Config.ipc_worker_size)
┃ [R] ↘ Do not use parentheses when defining a function which has no arguments.
┃       lib/ethereumex/config.ex:62 #(Ethereumex.Config.ipc_path)
┃ [R] ↘ Do not use parentheses when defining a function which has no arguments.
┃       lib/ethereumex/config.ex:57 #(Ethereumex.Config.client_type)
┃ [R] ↘ Do not use parentheses when defining a function which has no arguments.
┃       lib/ethereumex/config.ex:52 #(Ethereumex.Config.http_options)
┃ [R] ↘ Do not use parentheses when defining a function which has no arguments.
┃       lib/ethereumex/config.ex:37 #(Ethereumex.Config.rpc_url)
┃ [R] ↘ Do not use parentheses when defining a function which has no arguments.
┃       lib/ethereumex/config.ex:27 #(Ethereumex.Config.poolboy_config)
```

Reply:
Why do we need parentheses?

Reply:
1. It was a matter of choice between fixing 5 issues or 10 :) haha

2. I personally prefer parentheses because functions look weird without it and no such thing exists in erlang. And to keep the consistent look between functions. :)

Reply:
It's more conventional to skip parentheses so let's revet these changes
https://github.com/phoenixframework/phoenix/blob/master/lib/phoenix.ex

Reply:
Depends what you're reading:
https://github.com/lexmag/elixir-style-guide/blob/master/README.md#fun-parens
devs from @elixir-lang core team use that. 

Elixir Logger uses it:
https://github.com/elixir-lang/elixir/blob/master/lib/logger/lib/logger.ex#L580
Elixir ExUnit doesn't use it:
https://github.com/elixir-lang/elixir/blob/master/lib/ex_unit/lib/ex_unit.ex#L297-L306

Reply:
Ok. Please bump version, add a changelog and let's merge this